src/ocean/io/select/EpollSelectDispatcher.d

```diff
@@ -740,10 +787,17 @@ public class EpollSelectDispatcher : IEpollSelectDispatcherInfo
         bool caller_work_pending = false;
 
         if (select_cycle_hook !is null)
-            caller_work_pending = select_cycle_hook();
+            select_cycle_hooks ~= select_cycle_hook;
+
+        foreach (hook; select_cycle_hooks)
+        {
+            caller_work_pending |= hook();
+        }
 
         while ( this.registered_clients.length && !this.shutdown_triggered )
         {
+            caller_work_pending = false;
```

Comment:
This has to be set to false after the select call or otherwise select will always block.